From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dolphin XD <dolphinxd@users.noreply.github.com>
Date: Thu, 8 Oct 2026 01:55:08 +0000
Subject: [PATCH 3/3] Theme Dolphin's screens in the game's colors

Dolphin's light and dark palettes point at the xd_dolphin_* colors that
each game defines, so the settings screens and in-game menu match the
app's main menu (purple/red for Dolphin XD, desert tones for Dolphin
Colosseum).
---
 .../main/res/values-night/dolphin_colors.xml  | 56 +++++++++----------
 .../src/main/res/values/dolphin_colors.xml    | 56 +++++++++----------
 2 files changed, 56 insertions(+), 56 deletions(-)

diff --git a/Source/Android/app/src/main/res/values-night/dolphin_colors.xml b/Source/Android/app/src/main/res/values-night/dolphin_colors.xml
index 39e5ccc..44186ea 100644
--- a/Source/Android/app/src/main/res/values-night/dolphin_colors.xml
+++ b/Source/Android/app/src/main/res/values-night/dolphin_colors.xml
@@ -1,31 +1,31 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <!-- Generated by https://material-foundation.github.io/material-theme-builder/ -->
-    <color name="dolphin_primary">#9ECAFF</color>
-    <color name="dolphin_onPrimary">#003258</color>
-    <color name="dolphin_primaryContainer">#00497D</color>
-    <color name="dolphin_onPrimaryContainer">#D1E4FF</color>
-    <color name="dolphin_secondary">#BBC7DB</color>
-    <color name="dolphin_onSecondary">#253140</color>
-    <color name="dolphin_secondaryContainer">#3B4858</color>
-    <color name="dolphin_onSecondaryContainer">#D7E3F7</color>
-    <color name="dolphin_tertiary">#D6BEE4</color>
-    <color name="dolphin_onTertiary">#3B2948</color>
-    <color name="dolphin_tertiaryContainer">#523F5F</color>
-    <color name="dolphin_onTertiaryContainer">#F2DAFF</color>
-    <color name="dolphin_error">#FFB4AB</color>
-    <color name="dolphin_errorContainer">#93000A</color>
-    <color name="dolphin_onError">#690005</color>
-    <color name="dolphin_onErrorContainer">#FFDAD6</color>
-    <color name="dolphin_background">#1A1C1E</color>
-    <color name="dolphin_onBackground">#E2E2E6</color>
-    <color name="dolphin_surface">#1A1C1E</color>
-    <color name="dolphin_onSurface">#E2E2E6</color>
-    <color name="dolphin_surfaceVariant">#43474E</color>
-    <color name="dolphin_onSurfaceVariant">#C3C7CF</color>
-    <color name="dolphin_outline">#8D9199</color>
-    <color name="dolphin_inverseOnSurface">#1A1C1E</color>
-    <color name="dolphin_inverseSurface">#E2E2E6</color>
-    <color name="dolphin_inversePrimary">#0061A4</color>
-    <color name="dolphin_shadow">#000000</color>
+    <!-- Dolphin XD: the palette comes from the game being built (games/<game>/.../game_colors.xml). -->
+    <color name="dolphin_primary">@color/xd_dolphin_primary</color>
+    <color name="dolphin_onPrimary">@color/xd_dolphin_onPrimary</color>
+    <color name="dolphin_primaryContainer">@color/xd_dolphin_primaryContainer</color>
+    <color name="dolphin_onPrimaryContainer">@color/xd_dolphin_onPrimaryContainer</color>
+    <color name="dolphin_secondary">@color/xd_dolphin_secondary</color>
+    <color name="dolphin_onSecondary">@color/xd_dolphin_onSecondary</color>
+    <color name="dolphin_secondaryContainer">@color/xd_dolphin_secondaryContainer</color>
+    <color name="dolphin_onSecondaryContainer">@color/xd_dolphin_onSecondaryContainer</color>
+    <color name="dolphin_tertiary">@color/xd_dolphin_tertiary</color>
+    <color name="dolphin_onTertiary">@color/xd_dolphin_onTertiary</color>
+    <color name="dolphin_tertiaryContainer">@color/xd_dolphin_tertiaryContainer</color>
+    <color name="dolphin_onTertiaryContainer">@color/xd_dolphin_onTertiaryContainer</color>
+    <color name="dolphin_error">@color/xd_dolphin_error</color>
+    <color name="dolphin_errorContainer">@color/xd_dolphin_errorContainer</color>
+    <color name="dolphin_onError">@color/xd_dolphin_onError</color>
+    <color name="dolphin_onErrorContainer">@color/xd_dolphin_onErrorContainer</color>
+    <color name="dolphin_background">@color/xd_dolphin_background</color>
+    <color name="dolphin_onBackground">@color/xd_dolphin_onBackground</color>
+    <color name="dolphin_surface">@color/xd_dolphin_surface</color>
+    <color name="dolphin_onSurface">@color/xd_dolphin_onSurface</color>
+    <color name="dolphin_surfaceVariant">@color/xd_dolphin_surfaceVariant</color>
+    <color name="dolphin_onSurfaceVariant">@color/xd_dolphin_onSurfaceVariant</color>
+    <color name="dolphin_outline">@color/xd_dolphin_outline</color>
+    <color name="dolphin_inverseOnSurface">@color/xd_dolphin_inverseOnSurface</color>
+    <color name="dolphin_inverseSurface">@color/xd_dolphin_inverseSurface</color>
+    <color name="dolphin_inversePrimary">@color/xd_dolphin_inversePrimary</color>
+    <color name="dolphin_shadow">@color/xd_dolphin_shadow</color>
 </resources>
diff --git a/Source/Android/app/src/main/res/values/dolphin_colors.xml b/Source/Android/app/src/main/res/values/dolphin_colors.xml
index eddde11..29bdea1 100644
--- a/Source/Android/app/src/main/res/values/dolphin_colors.xml
+++ b/Source/Android/app/src/main/res/values/dolphin_colors.xml
@@ -1,33 +1,33 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <!-- Generated by https://material-foundation.github.io/material-theme-builder/ -->
-    <color name="dolphin_primary">#2196f3</color>
-    <color name="dolphin_onPrimary">#FFFFFF</color>
-    <color name="dolphin_primaryContainer">#D1E4FF</color>
-    <color name="dolphin_onPrimaryContainer">#001D36</color>
-    <color name="dolphin_secondary">#535F70</color>
-    <color name="dolphin_onSecondary">#FFFFFF</color>
-    <color name="dolphin_secondaryContainer">#D7E3F7</color>
-    <color name="dolphin_onSecondaryContainer">#101C2B</color>
-    <color name="dolphin_tertiary">#6B5778</color>
-    <color name="dolphin_onTertiary">#FFFFFF</color>
-    <color name="dolphin_tertiaryContainer">#F2DAFF</color>
-    <color name="dolphin_onTertiaryContainer">#251431</color>
-    <color name="dolphin_error">#BA1A1A</color>
-    <color name="dolphin_errorContainer">#FFDAD6</color>
-    <color name="dolphin_onError">#FFFFFF</color>
-    <color name="dolphin_onErrorContainer">#410002</color>
-    <color name="dolphin_background">#FDFCFF</color>
-    <color name="dolphin_onBackground">#1A1C1E</color>
-    <color name="dolphin_surface">#FDFCFF</color>
-    <color name="dolphin_onSurface">#1A1C1E</color>
-    <color name="dolphin_surfaceVariant">#DFE2EB</color>
-    <color name="dolphin_onSurfaceVariant">#43474E</color>
-    <color name="dolphin_outline">#73777F</color>
-    <color name="dolphin_inverseOnSurface">#F1F0F4</color>
-    <color name="dolphin_inverseSurface">#2F3033</color>
-    <color name="dolphin_inversePrimary">#9ECAFF</color>
-    <color name="dolphin_shadow">#000000</color>
+    <!-- Dolphin XD: the palette comes from the game being built (games/<game>/.../game_colors.xml). -->
+    <color name="dolphin_primary">@color/xd_dolphin_primary</color>
+    <color name="dolphin_onPrimary">@color/xd_dolphin_onPrimary</color>
+    <color name="dolphin_primaryContainer">@color/xd_dolphin_primaryContainer</color>
+    <color name="dolphin_onPrimaryContainer">@color/xd_dolphin_onPrimaryContainer</color>
+    <color name="dolphin_secondary">@color/xd_dolphin_secondary</color>
+    <color name="dolphin_onSecondary">@color/xd_dolphin_onSecondary</color>
+    <color name="dolphin_secondaryContainer">@color/xd_dolphin_secondaryContainer</color>
+    <color name="dolphin_onSecondaryContainer">@color/xd_dolphin_onSecondaryContainer</color>
+    <color name="dolphin_tertiary">@color/xd_dolphin_tertiary</color>
+    <color name="dolphin_onTertiary">@color/xd_dolphin_onTertiary</color>
+    <color name="dolphin_tertiaryContainer">@color/xd_dolphin_tertiaryContainer</color>
+    <color name="dolphin_onTertiaryContainer">@color/xd_dolphin_onTertiaryContainer</color>
+    <color name="dolphin_error">@color/xd_dolphin_error</color>
+    <color name="dolphin_errorContainer">@color/xd_dolphin_errorContainer</color>
+    <color name="dolphin_onError">@color/xd_dolphin_onError</color>
+    <color name="dolphin_onErrorContainer">@color/xd_dolphin_onErrorContainer</color>
+    <color name="dolphin_background">@color/xd_dolphin_background</color>
+    <color name="dolphin_onBackground">@color/xd_dolphin_onBackground</color>
+    <color name="dolphin_surface">@color/xd_dolphin_surface</color>
+    <color name="dolphin_onSurface">@color/xd_dolphin_onSurface</color>
+    <color name="dolphin_surfaceVariant">@color/xd_dolphin_surfaceVariant</color>
+    <color name="dolphin_onSurfaceVariant">@color/xd_dolphin_onSurfaceVariant</color>
+    <color name="dolphin_outline">@color/xd_dolphin_outline</color>
+    <color name="dolphin_inverseOnSurface">@color/xd_dolphin_inverseOnSurface</color>
+    <color name="dolphin_inverseSurface">@color/xd_dolphin_inverseSurface</color>
+    <color name="dolphin_inversePrimary">@color/xd_dolphin_inversePrimary</color>
+    <color name="dolphin_shadow">@color/xd_dolphin_shadow</color>
 
     <color name="dolphin_blue">#2196f3</color>
 
